apps/settings/test/unit/panels/call_barring/call_barring_test.js

```diff
@@ -0,0 +1,393 @@
+/* global MockMobileconnection */
+'use strict';
+
+require('/shared/test/unit/mocks/mock_navigator_moz_mobile_connections.js');
+
+suite('Call Barring settings', function() {
+  var realMozMobileConnection;
+
+  var _mobileConnection,
+      _serviceClass;
+
+  var callBarring;
+
+  suiteSetup(function(done) {
+
+    realMozMobileConnection = navigator.mozMobileConnection;
+    navigator.mozMobileConnection = MockMobileconnection;
+
+    var modules = [
+      'panels/call_barring/call_barring'
+    ];
+
+    var maps = {
+      'panels/call_barring/call_barring': {
+        'modules/settings_service': 'unit/mock_settings_service'
+      }
+    };
+
+    testRequire(modules, maps, function(CallBarring) {
+      callBarring = CallBarring;
+      done();
+    });
+
+  });
+
+  suiteTeardown(function() {
+    navigator.mozMobileConnection = realMozMobileConnection;
+  });
+
+  setup(function() {
+    _mobileConnection = navigator.mozMobileConnection();
```

Comment:
oh, i see. Then it would make more sense simply using `_mobileConnection = MockMobileConnection()` without setting `navigator.mozMobileConnection`. Note that `navigator.mozMobileConnection` does not exist anymore.